Tasks/Kubernetes/src/clusterconnection.ts

```diff
@@ -24,8 +24,21 @@ export default class ClusterConnection {
         {
             tl.debug(tl.loc("DownloadingClient"));
             this.kubectlPath = path.join(this.userDir, "kubectl") + this.getExecutableExtention();
-            var version = await utils.getStableKubectlVersion();
-            await utils.downloadKubectl(version, this.kubectlPath);
+            let versionOrLocation = tl.getInput("versionOrLocation");
```

Comment:
Let's move this logic to a function GetKubectlPath which take the inputs and return kubectl path.